changes/tasks.py

```diff
@@ -1468,6 +1469,78 @@ def run(self, vumi_message_id: str, event_type: str, **kwargs) -> None:
 process_whatsapp_system_event = ProcessWhatsAppSystemEvent()
 
 
+class ProcessWhatsAppTimeoutSystemEvent(Task):
+    """
+    Notifies a user that a message we send them on Whatsapp has not been delivered
+    within 7 days, she will be sent an outbound
+    message informing them and give the option to switch to SMS.
+    """
+
+    name = "ndoh_hub.changes.tasks.process_whatsapp_timeout_system_event"
+
+    def send_outbound(self, user_id, source_id, identity_uuid):
+        identity = is_client.get_identity(identity_uuid)
+        # Transform to django language code
+        for item in identity["results"]:
+            details = item["details"]
+            if "lang_code" in details:
+                language = details["lang_code"].lower().replace("_", "-")
+                with translation.override(language):
+                    text = translation.ugettext(
+                        "We see that your MomConnect WhatsApp messages are not being "
+                        "delivered. If you would like to receive your messages over "
+                        "SMS, reply ‘SMS’."
+                    )
+
+                utils.ms_client.create_outbound(
+                    {
+                        "to_identity": identity_uuid,
+                        "content": text,
+                        "channel": "JUNE_TEXT",
+                        "metadata": {},
+                    }
+                )
+
+    def handle_undelivered(self, user_id, source_id, identity_uuid):
+        identity = is_client.get_identity(identity_uuid)
+        for item in identity["results"]:
+            details = item["details"]
+            if "timeout_timestamp" not in details:
+                details["timeout_timestamp"] = time.mktime(datetime.now().timetuple())
```

Comment:
using `.timetuple()` is easier that `.timestamp(), because it rounds of the milliseconds of the timestamp i.e.  a .timestamp() will create a timestamp of 1543932583.366118 and 1543932583.370697 within seconds, while .tuple() will create 1543932483.0. Should I still change it?

Reply:
Yes I don't think that there's any harm in having more accuracy. It's not super important for this case, but it does make the code a bit easier to read, which is important